lib/Backend/BackwardPass.cpp

```diff
@@ -2735,6 +2848,168 @@ BackwardPass::ProcessBlock(BasicBlock * block)
 #endif
 }
 
+bool 
+BackwardPass::CanDeadStoreInstrForScopeObjRemoval(Sym *sym) const
+{
+    if (tag == Js::DeadStorePhase && this->currentInstr->IsStackArgsEnabled())
+    {
+        Func * currFunc = this->currentInstr->m_func;
+        switch (this->currentInstr->m_opcode)
+        {
+            case Js::OpCode::LdHeapArguments:
+            case Js::OpCode::LdHeapArgsCached:
+            case Js::OpCode::LdLetHeapArguments:
+            case Js::OpCode::LdLetHeapArgsCached:
+            {
```

Comment:
Actually the instr itself doesn't get dead stored, but it helps for not marking the sym of the LdHeapArguments sym from getting marked as SymUsed

---
In reply to: [68114689](https://github.com/Microsoft/ChakraCore/pull/1135#discussion_r68114689) [](ancestors = 68114689)